src/MicrosoftTeams.ts

```diff
@@ -1200,7 +1200,9 @@ export namespace authentication {
       frameContexts.task
     );
 
-    if (hostClientType === HostClientType.desktop) {
+    if (hostClientType === HostClientType.desktop
```

Comment:
Or it could be 
!!hostClientType && hostClientType !== HostClientType.web

Reply:
I like the way you have it. I think it's better to be explicit here since the else case uses vanilla window.open and can work in any browser environment whereas the android/ios/desktop case requires the host to handle most of the auth flow logic and would fail in any other environment.